tests/MongoDB.Driver.Core.Tests/Core/Clusters/ClusterTests.cs

```diff
@@ -481,6 +481,65 @@ public void DescriptionChanged_should_be_raised_when_the_description_changes()
             _capturedEvents.Any().Should().BeFalse();
         }
 
+        [Theory]
+        [ParameterAttributeData]
+        public void SelectServer_should_call_custom_selector(
+            [Values(true, false)] bool doNotFilterInitialServers,
```

Comment:
I suggest renaming `doNotFilterInitialServers` to `withEligibleServers`.

A negative in a variable name can be confusing.

Reply:
Done.